packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -723,37 +787,101 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
 
     // Proposes new price of True for relay event associated with `customAncillaryData` to optimistic oracle. If anyone
     // disagrees with the relay parameters and whether they map to an L2 deposit, they can dispute with the oracle.
-    function _requestAndProposeOraclePriceRelay(
-        uint256 amount,
+    function _requestProposeDispute(
+        address proposer,
+        address disputer,
+        uint256 proposerBond,
+        uint256 finalFee,
         uint32 requestTimestamp,
         bytes memory customAncillaryData
     ) private {
-        // Compute total proposal bond and pull from caller so that the OptimisticOracle can pull it from here.
-        uint256 finalFee = store.computeFinalFee(address(l1Token)).rawValue;
-        uint256 totalBond = (uint256(proposerBondPct) * amount) / 1e18 + finalFee;
+        uint256 totalBond = finalFee + proposerBond;
+        l1Token.safeApprove(address(optimisticOracle), totalBond);
+        try
+            optimisticOracle.requestAndProposePriceFor(
+                identifier,
+                requestTimestamp,
+                customAncillaryData,
+                IERC20(l1Token),
+                // Set reward to 0, since we'll settle proposer reward payouts directly from this contract after a relay
+                // proposal has passed the challenge period.
+                0,
+                // Set the Optimistic oracle proposer bond for the price request.
+                proposerBond,
+                // Set the Optimistic oracle liveness for the price request.
+                uint256(optimisticOracleLiveness),
+                proposer,
+                // Canonical value representing "True"; i.e. the proposed relay is valid.
+                int256(1e18)
+            )
+        returns (uint256 bondSpent) {
+            if (bondSpent < totalBond) {
+                // If the OO pulls less (due to a change in final fee), refund the proposer.
+                uint256 refund = totalBond - bondSpent;
+                l1Token.safeTransfer(proposer, refund);
+                l1Token.safeApprove(address(optimisticOracle), 0);
+            }
+        } catch {
+            // If there's an error in the OO, this means something has changed to make this request undisputable.
+            // To ensure the request does not go through by default, refund the prposer and return early, allowing
+            // the calling method to delete the request, but with no additional recourse by the OO.
+            l1Token.safeTransfer(proposer, totalBond);
```

Comment:
Do we want to emit a particular error in the catch case?

Reply:
like emit an error in the catch case to indicate that the happy path did not happen.

Reply:
you could emit something in the non-catch case (happy path) and then emit a separate require in the event it reverts. this way we can detect if something undesirable happened. otherwise, it's somewhat hard to figure this out

Reply:
it would be a standard event but just something named to indicate that the desired flow did not happen.